# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/PackedGLEnums_autogen.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/common/PackedGLEnums_autogen.h
--- a/gfx/angle/checkout/src/common/PackedGLEnums_autogen.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/common/PackedGLEnums_autogen.h	Fri Nov 08 17:06:00 2019 +0000
@@ -17,6 +17,7 @@
 #include <angle_gl.h>
 
 #include <cstdint>
+#include <ostream>
 
 namespace gl
 {
@@ -42,6 +43,7 @@
 template <>
 AlphaTestFunc FromGLenum<AlphaTestFunc>(GLenum from);
 GLenum ToGLenum(AlphaTestFunc from);
+std::ostream &operator<<(std::ostream &os, AlphaTestFunc value);
 
 enum class BufferBinding : uint8_t
 {
@@ -65,6 +67,7 @@
 template <>
 BufferBinding FromGLenum<BufferBinding>(GLenum from);
 GLenum ToGLenum(BufferBinding from);
+std::ostream &operator<<(std::ostream &os, BufferBinding value);
 
 enum class BufferUsage : uint8_t
 {
@@ -85,6 +88,7 @@
 template <>
 BufferUsage FromGLenum<BufferUsage>(GLenum from);
 GLenum ToGLenum(BufferUsage from);
+std::ostream &operator<<(std::ostream &os, BufferUsage value);
 
 enum class ClientVertexArrayType : uint8_t
 {
@@ -101,6 +105,7 @@
 template <>
 ClientVertexArrayType FromGLenum<ClientVertexArrayType>(GLenum from);
 GLenum ToGLenum(ClientVertexArrayType from);
+std::ostream &operator<<(std::ostream &os, ClientVertexArrayType value);
 
 enum class CullFaceMode : uint8_t
 {
@@ -115,6 +120,7 @@
 template <>
 CullFaceMode FromGLenum<CullFaceMode>(GLenum from);
 GLenum ToGLenum(CullFaceMode from);
+std::ostream &operator<<(std::ostream &os, CullFaceMode value);
 
 enum class FilterMode : uint8_t
 {
@@ -131,6 +137,7 @@
 template <>
 FilterMode FromGLenum<FilterMode>(GLenum from);
 GLenum ToGLenum(FilterMode from);
+std::ostream &operator<<(std::ostream &os, FilterMode value);
 
 enum class FogMode : uint8_t
 {
@@ -145,6 +152,36 @@
 template <>
 FogMode FromGLenum<FogMode>(GLenum from);
 GLenum ToGLenum(FogMode from);
+std::ostream &operator<<(std::ostream &os, FogMode value);
+
+enum class GraphicsResetStatus : uint8_t
+{
+    NoError              = 0,
+    GuiltyContextReset   = 1,
+    InnocentContextReset = 2,
+    UnknownContextReset  = 3,
+
+    InvalidEnum = 4,
+    EnumCount   = 4,
+};
+
+template <>
+GraphicsResetStatus FromGLenum<GraphicsResetStatus>(GLenum from);
+GLenum ToGLenum(GraphicsResetStatus from);
+std::ostream &operator<<(std::ostream &os, GraphicsResetStatus value);
+
+enum class HandleType : uint8_t
+{
+    OpaqueFd = 0,
+
+    InvalidEnum = 1,
+    EnumCount   = 1,
+};
+
+template <>
+HandleType FromGLenum<HandleType>(GLenum from);
+GLenum ToGLenum(HandleType from);
+std::ostream &operator<<(std::ostream &os, HandleType value);
 
 enum class HintSetting : uint8_t
 {
@@ -159,6 +196,29 @@
 template <>
 HintSetting FromGLenum<HintSetting>(GLenum from);
 GLenum ToGLenum(HintSetting from);
+std::ostream &operator<<(std::ostream &os, HintSetting value);
+
+enum class ImageLayout : uint8_t
+{
+    Undefined                      = 0,
+    General                        = 1,
+    ColorAttachment                = 2,
+    DepthStencilAttachment         = 3,
+    DepthStencilReadOnlyAttachment = 4,
+    ShaderReadOnly                 = 5,
+    TransferSrc                    = 6,
+    TransferDst                    = 7,
+    DepthReadOnlyStencilAttachment = 8,
+    DepthAttachmentStencilReadOnly = 9,
+
+    InvalidEnum = 10,
+    EnumCount   = 10,
+};
+
+template <>
+ImageLayout FromGLenum<ImageLayout>(GLenum from);
+GLenum ToGLenum(ImageLayout from);
+std::ostream &operator<<(std::ostream &os, ImageLayout value);
 
 enum class LightParameter : uint8_t
 {
@@ -181,6 +241,7 @@
 template <>
 LightParameter FromGLenum<LightParameter>(GLenum from);
 GLenum ToGLenum(LightParameter from);
+std::ostream &operator<<(std::ostream &os, LightParameter value);
 
 enum class LogicalOperation : uint8_t
 {
@@ -208,6 +269,7 @@
 template <>
 LogicalOperation FromGLenum<LogicalOperation>(GLenum from);
 GLenum ToGLenum(LogicalOperation from);
+std::ostream &operator<<(std::ostream &os, LogicalOperation value);
 
 enum class MaterialParameter : uint8_t
 {
@@ -225,6 +287,7 @@
 template <>
 MaterialParameter FromGLenum<MaterialParameter>(GLenum from);
 GLenum ToGLenum(MaterialParameter from);
+std::ostream &operator<<(std::ostream &os, MaterialParameter value);
 
 enum class MatrixType : uint8_t
 {
@@ -239,6 +302,7 @@
 template <>
 MatrixType FromGLenum<MatrixType>(GLenum from);
 GLenum ToGLenum(MatrixType from);
+std::ostream &operator<<(std::ostream &os, MatrixType value);
 
 enum class PointParameter : uint8_t
 {
@@ -254,8 +318,9 @@
 template <>
 PointParameter FromGLenum<PointParameter>(GLenum from);
 GLenum ToGLenum(PointParameter from);
+std::ostream &operator<<(std::ostream &os, PointParameter value);
 
-enum class ProvokingVertex : uint8_t
+enum class ProvokingVertexConvention : uint8_t
 {
     FirstVertexConvention = 0,
     LastVertexConvention  = 1,
@@ -265,8 +330,9 @@
 };
 
 template <>
-ProvokingVertex FromGLenum<ProvokingVertex>(GLenum from);
-GLenum ToGLenum(ProvokingVertex from);
+ProvokingVertexConvention FromGLenum<ProvokingVertexConvention>(GLenum from);
+GLenum ToGLenum(ProvokingVertexConvention from);
+std::ostream &operator<<(std::ostream &os, ProvokingVertexConvention value);
 
 enum class QueryType : uint8_t
 {
@@ -285,6 +351,7 @@
 template <>
 QueryType FromGLenum<QueryType>(GLenum from);
 GLenum ToGLenum(QueryType from);
+std::ostream &operator<<(std::ostream &os, QueryType value);
 
 enum class ShaderType : uint8_t
 {
@@ -300,6 +367,7 @@
 template <>
 ShaderType FromGLenum<ShaderType>(GLenum from);
 GLenum ToGLenum(ShaderType from);
+std::ostream &operator<<(std::ostream &os, ShaderType value);
 
 enum class ShadingModel : uint8_t
 {
@@ -313,6 +381,7 @@
 template <>
 ShadingModel FromGLenum<ShadingModel>(GLenum from);
 GLenum ToGLenum(ShadingModel from);
+std::ostream &operator<<(std::ostream &os, ShadingModel value);
 
 enum class TextureCombine : uint8_t
 {
@@ -332,6 +401,7 @@
 template <>
 TextureCombine FromGLenum<TextureCombine>(GLenum from);
 GLenum ToGLenum(TextureCombine from);
+std::ostream &operator<<(std::ostream &os, TextureCombine value);
 
 enum class TextureEnvMode : uint8_t
 {
@@ -349,6 +419,7 @@
 template <>
 TextureEnvMode FromGLenum<TextureEnvMode>(GLenum from);
 GLenum ToGLenum(TextureEnvMode from);
+std::ostream &operator<<(std::ostream &os, TextureEnvMode value);
 
 enum class TextureEnvParameter : uint8_t
 {
@@ -379,6 +450,7 @@
 template <>
 TextureEnvParameter FromGLenum<TextureEnvParameter>(GLenum from);
 GLenum ToGLenum(TextureEnvParameter from);
+std::ostream &operator<<(std::ostream &os, TextureEnvParameter value);
 
 enum class TextureEnvTarget : uint8_t
 {
@@ -392,6 +464,7 @@
 template <>
 TextureEnvTarget FromGLenum<TextureEnvTarget>(GLenum from);
 GLenum ToGLenum(TextureEnvTarget from);
+std::ostream &operator<<(std::ostream &os, TextureEnvTarget value);
 
 enum class TextureOp : uint8_t
 {
@@ -407,6 +480,7 @@
 template <>
 TextureOp FromGLenum<TextureOp>(GLenum from);
 GLenum ToGLenum(TextureOp from);
+std::ostream &operator<<(std::ostream &os, TextureOp value);
 
 enum class TextureSrc : uint8_t
 {
@@ -422,6 +496,7 @@
 template <>
 TextureSrc FromGLenum<TextureSrc>(GLenum from);
 GLenum ToGLenum(TextureSrc from);
+std::ostream &operator<<(std::ostream &os, TextureSrc value);
 
 enum class TextureTarget : uint8_t
 {
@@ -446,6 +521,7 @@
 template <>
 TextureTarget FromGLenum<TextureTarget>(GLenum from);
 GLenum ToGLenum(TextureTarget from);
+std::ostream &operator<<(std::ostream &os, TextureTarget value);
 
 enum class TextureType : uint8_t
 {
@@ -465,6 +541,7 @@
 template <>
 TextureType FromGLenum<TextureType>(GLenum from);
 GLenum ToGLenum(TextureType from);
+std::ostream &operator<<(std::ostream &os, TextureType value);
 
 enum class VertexArrayType : uint8_t
 {
@@ -481,6 +558,7 @@
 template <>
 VertexArrayType FromGLenum<VertexArrayType>(GLenum from);
 GLenum ToGLenum(VertexArrayType from);
+std::ostream &operator<<(std::ostream &os, VertexArrayType value);
 
 enum class WrapMode : uint8_t
 {
@@ -496,6 +574,7 @@
 template <>
 WrapMode FromGLenum<WrapMode>(GLenum from);
 GLenum ToGLenum(WrapMode from);
+std::ostream &operator<<(std::ostream &os, WrapMode value);
 
 }  // namespace gl
 